test/remote/gateways/remote_nmi_test.rb

```diff
@@ -45,9 +46,50 @@ def test_invalid_login
     assert_equal 'Authentication Failed', response.message
   end
 
+  def test_invalid_login_segurity_key_empty
+    gateway_secure = NmiGateway.new(security_key: '')
+    assert response = gateway_secure.purchase(@amount, @credit_card, @options)
+    assert_failure response
+    assert_equal 'Authentication Failed', response.message
+  end
+
+  def test_valid_login_uname_password
+    @gateway = NmiGateway.new(login: 'demo', password: 'password')
+    assert response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+  end
+
+  def test_valid_login_segurity_key
+    gateway_secure = NmiGateway.new(fixtures(:nmi_secure))
+    assert response = gateway_secure.purchase(@amount, @credit_card, @options)
+    assert_success response
+  end
+
+  def test_successful_authorization_segurity_key
```

Comment:
❓  Question:

You are just dealing with Authorization credentials on this PR, why are you testing authorize/purchase transactions ?, they are already covered by other test cases.